frontend/packages/dev-console/src/components/gitops/utils/useTransformedEnvsData.tsx

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { GitOpsResource, WORKLOAD_KINDS, GitOpsEnvironment } from './gitops-types';
+import { routeDecoratorIcon } from '../../import/render-utils';
+import PodRingWrapper from '../details/PodRingWrapper';
+import CommitDetails from '../details/CommitDetails';
+import TimestampWrapper from '../details/TimestampWrapper';
+
+const getWorkLoad = (resources: GitOpsResource[]) => {
+  return _.find(resources, (res) => _.includes(WORKLOAD_KINDS, res.kind));
+};
+
+const getTransformedServices = (env: GitOpsEnvironment) => {
+  return _.sortBy(
+    _.map(env?.services, (service) => {
+      const workload = getWorkLoad(service?.resources);
+      const image = service?.images?.[0];
+      return {
+        ...service,
+        source: {
+          ...service?.source,
+          icon: routeDecoratorIcon(service?.source?.url, 12),
+        },
+        workloadKind: workload?.kind,
+        image,
+        podRing: <PodRingWrapper workload={workload} />,
+        commitDetails: <CommitDetails imageName={image} />,
+      };
+    }),
+    ['name'],
+  );
+};
+
+const getTransformedEnvsData = (envsData: GitOpsEnvironment[]) => {
+  return _.map(envsData, (env) => {
+    if (env) {
+      const resModels = _.flatten(_.map(env?.services, (service) => service.resources));
+      const timestamp = <TimestampWrapper resModels={resModels} />;
+      const services = getTransformedServices(env);
+      return {
+        ...env,
+        services,
+        timestamp,
+      };
+    }
+    return undefined;
+  });
+};
+
+const useTransformedEnvsData = (envsData: GitOpsEnvironment[]) =>
```

Comment:
No need to create a custom hook for this. Probably better to just use `React.useMemo` in component itself and just use the util function.